tt_eager/tt_dnn/queue_api.hpp

```diff
@@ -0,0 +1,79 @@
+void EnqueueAllocateDeviceBuffer(Queue&, DeviceBuffer&);
+void EnqueueAllocateHostBuffer(Queue&, HostBuffer&);
+
+
+void EnqueueDeallocate(Queue&, Tensor&);
+void EnqueueReallocate(Queue&, Tensor&);
+
+
+void EnqueueHostToDeviceTransfer(Queue&, Tensor& src, Tensor& dst);
+void EnqueueDeviceToHostTransfer(Queue&, Tensor& src, Tensor& dst);
```

Comment:
I suggest:
_void EnqueueHostToDeviceTransfer(Queue&, Tensor& dst, void* src, size_t transfer_size);_
_void EnqueueDeviceToHostTransfer(Queue&, void* dst, Tensor& src, size_t src_offset, size_t transfer_size);_

Reply:
what would happen if the Tensor size mismatches `transfer_size`?

Reply:
In the case of a host-to-device, the transfer size will always match the Tensor size.

In the case of a device-to-host, the transfer size might not match the Tensor size. In this case, partial data of the Tensor should be read.